src/instrumentation.py

```diff
@@ -0,0 +1,124 @@
+from prometheus_client import CollectorRegistry, Counter, Gauge
+
+from src.constants import TASK_EVENT_LABELS, WORKER_EVENT_LABELS, LabelName
+
+registry = CollectorRegistry(auto_describe=True)
+
+
+class EventInstrumentMixin:
+    @property
+    def labelnames(self):
+        return self._labelnames
+
+    @property
+    def name(self):
+        return self._name
+
+    @property
+    def documentation(self):
+        return self._documentation
+
+
+class EventCounter(Counter, EventInstrumentMixin):
+    pass
+
+
+class EventGauge(Gauge, EventInstrumentMixin):
+    pass
+
+
+task_sent_event_counter = EventCounter(
```

Comment:
Since it's pretty much set in stone what counters/gauges we need I decided to make them here and reuse throughout the project.